ov/interconnect_type.go

```diff
@@ -0,0 +1,135 @@
+package ov
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/HewlettPackard/oneview-golang/rest"
+	"github.com/HewlettPackard/oneview-golang/utils"
+	"github.com/docker/machine/libmachine/log"
+)
+
+type InterconnectType struct {
+	Category                 string                 `json:"category,omitempty"`                 // "category": "interconnect-types",
+	Created                  string                 `json:"created,omitempty"`                  // "created": "20150831T154835.250Z",
+	Description              string                 `json:"description,omitempty"`              // "description": "Interconnect Type 1",
+	DownlinkCount            int                    `json:"downlinkCount,omitempty"`            // "downlinkCount": 2,
+	DownlinkPortCapability   DownlinkPortCapability `json:"downlinkPortCapability,omitempty"`   // "downlinkPortCapability": {...},
+	ETAG                     string                 `json:"eTag,omitempty"`                     // "eTag": "1441036118675/8",
+	InterconnectCapabilities InterconnectCapability `json:"interconnectCapabilities,omitempty"` // "interconnectCapabilities": {...},
+	MaximumFirmwareVersion   string                 `json:"maximumFirmwareVersion,omitempty"`   // "maximumFirmwareVersion": "3.0.0",
+	MinimumFirmwareVersion   string                 `json:"minimumFirmwareVersion,omitempty"`   // "minimumFirmwareVersion": "2.0.0",
+	Modified                 string                 `json:"modified,omitempty"`                 // "modified": "20150831T154835.250Z",
+	Name                     string                 `json:"name,omitempty"`                     // "name": null,
```

Comment:
it can be null if we look at the comments, if thats true we should use utils.Nstring